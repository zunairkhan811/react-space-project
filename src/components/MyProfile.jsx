import { useSelector } from "react-redux";
const MyProfile = () => {
    const data = useSelector((state)=>state.rockets)
    return(
        <div className="profile-container">
            {data.rockets
             .filter((item)=>item.reserved)
             .map((item)=>(
                <ul>
                    <li>{item.rocket_name}</li>
                </ul>
              ))
            
            }

        </div>
    )
};

export default MyProfile;
