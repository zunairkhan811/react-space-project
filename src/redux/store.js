/* eslint-disable import/prefer-default-export */
import { configureStore } from '@reduxjs/toolkit';
import MissionReducer from './mission/missionSlice';
import rocketsReducer from './rocket/rocketSlice';

export const store = configureStore({
  reducer: {
    missions: MissionReducer,
    rockets: rocketsReducer,
  },
});
