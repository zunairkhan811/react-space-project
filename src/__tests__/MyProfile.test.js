import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import MyProfile from '../components/MyProfile';
import { store } from '../redux/store';

it('Profile component renders correctly', () => {
  const tree = TestRenderer.create(
    <Provider store={store}>
      <MyProfile />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
