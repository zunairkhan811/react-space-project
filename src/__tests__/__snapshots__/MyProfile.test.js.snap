// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="profile-container"
>
  <div
    className="my-profile"
  >
    <div
      className="my-missions"
    >
      <h2
        className="missions-title"
      >
        My Missions
      </h2>
      <table
        className="missions-table"
      >
        <tbody />
      </table>
    </div>
  </div>
  <div
    className="rocket-container"
  >
    <h1>
      My Rockets
    </h1>
    <ul
      className="rockets-lists"
    />
  </div>
</div>
`;
